src/json-patch.ts

```diff
@@ -136,30 +189,29 @@ module jsonpatch {
       obj[key] = this.value;
       return removed;
     },
-    move: function (obj, key, tree) {
-      var getOriginalDestination: any = { op: "_get", path: this.path };
-      apply(tree, [getOriginalDestination]);
+    move: function (obj, key, document) {
+      const originalValue = jsonpatch.getValueByPointer(document, this.path);
       // In case value is moved up and overwrites its ancestor
-      var original = getOriginalDestination.value === undefined ?
-        undefined : JSON.parse(JSON.stringify(getOriginalDestination.value));
+      var original = originalValue === undefined ?
+        undefined : JSON.parse(JSON.stringify(originalValue));
```

Comment:
Shouldn't this use the clone helper? Why is move cloning? In my use case I don't care if the value is the same reference